### DISCUSSION TITLE
Support setting / customizing image in Toast or Snackbar

### DISCUSSION THREAD

**Discussion by MichaelLHerman** [CommunityToolkit/Maui#2378] [New Feature Discussions]
I have a design requirement for toast notifications which includes an image and a background color

<img width="900" alt="Screenshot 2024-12-11 at 11 33 32 AM" src="https://github.com/user-attachments/assets/7d1f4f63-2bb4-488e-996f-3af4aaf86b31" />

I'm currently using Mopups library for this, displaying a generated custom view in a Mopups PopupPage, but Mopups does have some random exceptions and I was considering moving to CommunityToolkit.Maui.Alerts

iOS already creates a custom view (PlatformSnackbar/PlatformToast) for toast and would be fairly easy to implement the ability to add an image.  
I assume the Snackbar and Toast APIs were created to mirror the Android Material Components apis which don't provide mechanism to add an image.   It seems like Android Toast is geared for when the app is in the background since it includes the app icon.   This isn't really documented in the Maui Toast API [documentation](https://learn.microsoft.com/en-us/dotnet/communitytoolkit/maui/alerts/toast?tabs=android) and may be unexpected/undesirable for someone coming from iOS and not Android.  (Asked before as #1200)  Snackbar does not seem to be a good replacement for Toast because it seems to leave extra space for the action button even when action is null and text is set to String.Empty

<img width="226" alt="Screenshot 2024-12-11 at 12 42 24 PM" src="https://github.com/user-attachments/assets/0d13d35d-c874-4276-ad9f-ec3bd6a25fe0" />

Maybe the Android implementation should be a custom view as well (extending BaseTransientBottomBar) to allow full customization - removing the app icon and allowing a custom icon.

**Comment by stephenquan** [comment]
You can also consider using [CommunityToolkit.Maui Popup](https://learn.microsoft.com/en-us/dotnet/communitytoolkit/maui/views/popup) with an [IDispatchTimer](https://learn.microsoft.com/en-us/dotnet/api/microsoft.maui.dispatching.idispatchertimer?view=net-maui-9.0).

**Comment by VladislavAntonyuk** [comment]
It seems like Android 35 adds the image automatically
